### ISSUE TITLE
Use taghelpers wherever possible 

### ISSUE THREAD

**Issue by petrsvihlik** [petrsvihlik/petrsvihlik.com#2]
to make the syntax uniform.

- https://taghelperpack.net/
- https://github.com/Kentico/kontent-aspnetcore

~~blocked by https://github.com/statiqdev/Statiq.Framework/issues/143~~

**Comment by daveaglick** [comment]
See the discussion in https://github.com/statiqdev/Statiq.Framework/issues/143 - it appears tag helpers (including custom tag helpers) are supported out of the box. Perhaps a `@addTagHelper` directive is needed to light them up? I also added a comment to that issue about the possibility of adding them via the module (we can continue the discussion there to keep everything together).

**Comment by petrsvihlik** [comment]
Closing as superseded by the architecture change in #59.

Tag helpers are an MVC Razor Views feature and aren't supported in Razor components, which is what the templates are now (rendered via ASP.NET Core's `HtmlRenderer`). The goal of this issue — uniform, HTML-like syntax instead of `@Html.Partial(...)` calls — is fulfilled by the component model itself: templates compose as `<SiteLayout>`, `<PostList Model="..." />`, `<PostArticle Article="..." />` (see `Components/`).

The referenced libraries are also dead ends today: TagHelperPack is MVC-only and dormant, the Kontent helpers are irrelevant since the content moved to Markdown, and the Statiq blocker is moot since Statiq was removed. The repo's only actual tag helper (`GistTagHelper`) turned out to be unreferenced by any template and was deleted in #56.

If the underlying want was nice syntax for rich embeds *inside markdown posts* (gists etc.), the equivalent in the current setup is a small Markdig extension rather than a tag helper — tracked better by #15.

---
_Generated by [Claude Code](https://claude.ai/code)_
